bridge/contracts/NativeERCBridgePoolBase.sol

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.16;
+import "contracts/interfaces/IBridgePool.sol";
+import "contracts/utils/ImmutableAuth.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "contracts/libraries/errors/NativeERCBridgePoolBaseErrors.sol";
+import "contracts/utils/AccusationsLibrary.sol";
+import "contracts/libraries/parsers/MerkleProofParserLibrary.sol";
+import "contracts/libraries/parsers/VSPreImageParserLibrary.sol";
+import "contracts/libraries/parsers/PClaimsParserLibrary.sol";
+import "contracts/utils/MerkleProofLibrary.sol";
+import "contracts/Snapshots.sol";
+
+/// @custom:salt NativeERCBridgePoolBase
+/// @custom:deploy-type deployUpgradeable
+abstract contract NativeERCBridgePoolBase is ImmutableSnapshots {
```

Comment:
I think you need to inherit IbridgePool here and implement the base functionality of withdraw and deposit in this abstract contract

Reply:
Done as requested